invenio_records_resources/records/systemfields/files/manager.py

```diff
@@ -249,6 +249,32 @@ def delete_all(self, remove_obj=True, softdelete_obj=True, remove_rf=False):
                 remove_rf=remove_rf,
             )
 
+    def full_teardown(self):
+        """Hard delete bucket and Object versions associated with file manager."""
+        if self.enabled:
+            object_version_ids = [
+                entry.object_version_id for entry in self.entries.values()
+            ]
+            ObjectVersion.remove_many(object_version_ids, self.bucket)
+        self.remove_bucket(force=True)
+
+    def teardown(self, full=True):
+        """Clean up all file manager related instances.
+
+        Deletes FileRecords, Bucket and ObjectVersions attached to file manager.
+        equivalent of:
+        softdelete_obj = false, remove_rf=True, remove_obj=true
+        option of deleting only FileRecords if full=False
+        """
+        file_record_ids = [entry.id for entry in self.entries.values()]
+        self.file_cls.remove_many(file_record_ids)
+
+        if full:
+            self.full_teardown()
```

Comment:
minor/nit: since `full_teardown` is only called here, I would just inline it to keep the logic all in one place.